lib/MCT/Controller/Presentation.pm

```diff
@@ -11,17 +11,74 @@ sub show {
   $c->delay(
     sub { $p->load(shift->begin) },
     sub {
-      $c->app->log->debug('about to render');
       my ($delay, $err) = @_;
       die $err if $err;
-      $p->in_storage ? $c->render('presentation/show', p => $p) : $c->reply->not_found;      
+      $p->in_storage ? $c->render('presentation/show', p => $p) : $c->reply->not_found;
     },
   );
 }
 
-sub save {
+sub edit {
   my $c = shift;
-   
+  return $c->render('presentation/edit')
+    unless my $url_name = $c->stash('url_name');
+
+  my $p = $c->model->presentation(
+    conference => $c->stash('conference')->identifier,
+    url_name   => $url_name,
+  );
+  $c->delay(
+    sub { $p->load(shift->begin) },
+    sub {
+      my ($delay, $err) = @_;
+      die $err if $err;
+      $c->render_not_authorized unless $p->user_can_update($c->session('username'));
+      $c->render('presentation/edit', p => $p);
+    },
+  );
+}
+
+sub store {
+  my $c = shift;
+
+  my $id = $c->param('id');
+  my $p = $c->model->presentation(
+    conference => $c->stash('conference')->identifier,
+    $id ? (id => $id) : (),
+  );
+
+  my $title  = $c->param('title');
+  my %set = (
+    abstract => $c->param('abstract'),
+    author   => $c->session('username'),
+    subtitle => $c->param('subtitle'),
+    title    => $title,
+    url_name => $c->param('url_name') || $c->_url_name($title),
```

Comment:
what if people want a specific one in the future? I'm willing to agree, but it's not hard to leave it as is.